### PR TITLE
feat: Add bytes affine

### PR DESCRIPTION
**Pull request by bsbds**
Based on: #656 

Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)

  Adds a new type `BytesAffine`, which defines the order of the keys. 
  FYI: 
- https://etcd.io/docs/v3.3/learning/api/#key-ranges
- https://github.com/etcd-io/etcd/blob/e68afe7eb7ce3e5f83deb2a8ad58faa3b35494b9/pkg/adt/interval_tree.go#L919-L947
* what changes does this pull request make?

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)

### CODE REVIEW

File: crates/xlineapi/src/interval.rs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+mod tests {
+    use super::*;
+
+    #[test]
+    fn bytes_affine_cmp_is_ok() {
+        assert_eq!(BytesAffine::new_key("abc"), BytesAffine::new_key("abc"));
+        assert!(BytesAffine::new_key("a") < BytesAffine::new_key("b"));
+        assert!(BytesAffine::new_key("abcd") < BytesAffine::new_key("b"));
+        assert!(BytesAffine::new_key("abcd") < BytesAffine::new_unbounded());
+        assert!(BytesAffine::new_key("123") < BytesAffine::new_unbounded());
```
Comment (Phoenix500526) [inline crates/xlineapi/src/interval.rs:73]: This line is duplicated. It doesn't provide more information with the previous one.

Comment (bsbds) [reply crates/xlineapi/src/interval.rs]: removed

File: crates/xlineapi/src/interval.rs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn bytes_affine_cmp_is_ok() {
```
Comment (Phoenix500526) [inline crates/xlineapi/src/interval.rs:68]: Please cover the `KeyRange` from trait in this test case.

**Review by Phoenix500526 — APPROVED** [review]
LGTM，please squash these commits into one.

**Review by rogercloud — APPROVED** [review]

**Pull request merged**